src/wp-includes/blocks/file.php

```diff
@@ -53,39 +36,22 @@ static function ( $matches ) {
 		$content
 	);
 
-	// If it uses the Interactivity API, add the directives.
-	if ( $should_load_view_script ) {
+	// If it's interactive, enqueue the script module and add the directives.
+	if ( ! empty( $attributes['displayPreview'] ) ) {
+		wp_enqueue_script_module( '@wordpress/block-library/file' );
+
 		$processor = new WP_HTML_Tag_Processor( $content );
 		$processor->next_tag();
-		$processor->set_attribute( 'data-wp-interactive', '' );
+		$processor->set_attribute( 'data-wp-interactive', '{"namespace":"core/file"}' );
 		$processor->next_tag( 'object' );
-		$processor->set_attribute( 'data-wp-bind--hidden', '!selectors.core.file.hasPdfPreview' );
+		$processor->set_attribute( 'data-wp-bind--hidden', '!state.hasPdfPreview' );
 		$processor->set_attribute( 'hidden', true );
 		return $processor->get_updated_html();
 	}
 
 	return $content;
 }
 
-/**
- * Ensure that the view script has the `wp-interactivity` dependency.
- *
- * @since 6.4.0
- *
- * @global WP_Scripts $wp_scripts
- */
-function block_core_file_ensure_interactivity_dependency() {
```

Comment:
cc @c4rl0sbr4v0

Reply:
Done in https://github.com/c4rl0sbr4v0/wordpress-develop/commit/a3eba9d677b871362a2df2fc85c0280a57d117dc

It seems that I cannot make PRs to your fork @youknowriad

Reply:
We should try and make any changes in the Gutenberg repo and then update this PR so that we have a clear audit trail. Otherwise we're going to have a lot of `Backport from WordPress Core` PRs which can become very confusing.

The _optimal_ flow is:

`Gutenberg -> WP Core`.